contracts/dtoken/src/supply.rs

```diff
@@ -68,8 +68,12 @@ impl Contract {
             }
         };
 
-        let exchange_rate: Balance =
-            self.get_exchange_rate((balance_of - Balance::from(token_amount)).into());
+        let exchange_rate: Balance = self.get_exchange_rate(
+            (balance_of - Balance::from(token_amount)).into(),
+            self.get_total_borrows(),
```

Comment:
@dmytro-shash get_total_borrows, get_total_reserves, get_total_supplies should be inside of `get_exchange_rate` otherwise it's the same as `calculate_exchange_rate`